### PR TITLE
BOM-1539 : Add python 3.8 to testing

### PR DESCRIPTION
**Pull request by mzulqarnain1**
Relevant JIRA : [https://openedx.atlassian.net/browse/BOM-1539](https://openedx.atlassian.net/browse/BOM-1539)

- Removed Python 2.7
- Added Python 3.8 to testing
- Removed Django versions < 2.2
- Ran make upgrade
- Fixed some quality failures

### CODE REVIEW

File: requirements/base.in [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Core requirements for using this application
 -c constraints.txt
 
-Django>=1.11                          # Web application framework
+Django>=2.2                          # Web application framework
```
Comment (awais786) [inline requirements/base.in:4]: django<3.0

**Comment by awais786** [comment]
@mzulqarnain1 in case of allow fails plz add new ticket for followup.

File: requirements/pip-tools.txt [hunk 2]
```diff
@@ -4,6 +4,9 @@
 #
 #    make upgrade
 #
-click==7.0                # via pip-tools
-pip-tools==4.4.0
+click==7.1.1              # via pip-tools
+pip-tools==5.0.0          # via -r requirements/pip-tools.in
```
Comment (awais786) [inline requirements/pip-tools.txt:8]:
````
# pip20+ requires for greater version of pip-tools. But installing 19.3.1 in all our containers and VMs.
pip-tools==4.5.1
```

File: requirements/base.in [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Core requirements for using this application
 -c constraints.txt
 
-Django>=1.11                          # Web application framework
+Django>=2.2,<3.0                          # Web application framework
```
Comment (UsamaSadiq) [inline requirements/base.in:4]: You should update the constraint to `django<2.3` and move the constraint in the `constraints` file.

File: requirements/dev.txt [hunk 4]
```diff
@@ -4,73 +4,70 @@
...
+isort==4.3.21             # via -r requirements/quality.txt, pylint
+jinja2==2.11.2            # via -r requirements/quality.txt, code-annotations
+lazy-object-proxy==1.4.3  # via -r requirements/quality.txt, astroid
+markupsafe==1.1.1         # via -r requirements/quality.txt, jinja2
+mccabe==0.6.1             # via -r requirements/quality.txt, pylint
+more-itertools==8.2.0     # via -r requirements/quality.txt, pytest
+packaging==20.3           # via -r requirements/quality.txt, -r requirements/travis.txt, caniusepython3, pytest, tox
+pathlib2==2.3.5           # via -r requirements/quality.txt, pytest
+pbr==5.4.5                # via -r requirements/quality.txt, stevedore
+pip-tools==5.0.0          # via -r requirements/pip-tools.txt
```
Comment (UsamaSadiq) [inline requirements/dev.txt:38]: Add in the `constraints` file.
`# pip20+ requires for greater version of pip-tools. But installing 19.3.1 in all our containers and VMs.`
`pip-tools==4.5.1`

File: tox.ini [hunk 5]
```diff
@@ -1,16 +1,14 @@
 [tox]
-envlist = py{27,35}-django{111},py{35}-django{20,21,22}
+envlist = py{35,38}-django{22,30}
```
Comment (jmbowman) [inline tox.ini:2]: Django 3.0 dropped support for Python 3.5, so that combination wouldn't work.

**Pull request merged**